pkg/apis/serving/v1beta1/inference_service_validation.go

```diff
@@ -303,6 +322,18 @@ func validateScalingHPACompExtension(compExtSpec *ComponentExtensionSpec) error
 	return nil
 }
 
+func validateScalingKedaCompExtension(compExtSpec *ComponentExtensionSpec) error {
+	metric := *compExtSpec.ScaleMetric
```

Comment:
if `compExtSpec.ScaleMetric` is nil, this line will run into nil pointer.
You need to give `metric` a default value, and then call 
```
if compExtSpec.ScaleMetric != nil {
		metric = *compExtSpec.ScaleMetric
	}
```